src/openforms/submissions/tests/factories.py

```diff
@@ -312,12 +312,19 @@ class Meta:
 
 
 class TemporaryFileUploadFactory(factory.django.DjangoModelFactory):
+    submission = factory.SubFactory(SubmissionFactory)
     file_name = factory.Faker("file_name")
     content = factory.django.FileField(filename="file.dat", data=b"content")
 
     class Meta:
         model = TemporaryFileUpload
 
+    class Params:
+        is_legacy = factory.Trait(
+            legacy=True,
+            submission=None,
+        )
```

Comment:
I recently learned about `factory.Maybe` :) so there's a way to automatically make this so if `legacy=True` is provided.